iOSVideoApp/iOSVideoApp/views/VideoPostsCollectionViewController.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  VideoPostsCollectionViewController.swift
+//  iOSVideoApp
+//
+//  Created by Ezra Black on 5/6/20.
+//  Copyright © 2020 Casanova Studios. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+
+private let reuseIdentifier = "Cell"
+
+class VideoPostsCollectionViewController: UICollectionViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Uncomment the following line to preserve selection between presentations
+        // self.clearsSelectionOnViewWillAppear = false
+
+        // Register cell classes
+        self.collectionView!.register(UICollectionViewCell.self, forCellWithReuseIdentifier: reuseIdentifier)
+
+        // Do any additional setup after loading the view.
+    }
+    
+    // MARK: - Properties
+    
+    private func requestPermissionAndShowCamera() {
+            let status = AVCaptureDevice.authorizationStatus(for: .video)
+            
+            switch status {
+            case .notDetermined: // first time we've requested access
+                requestPermission()
+            case .restricted: // parental controls prevent user from camera/mic
+                fatalError("Tell user they need to request permission from parent")
+            case .denied:
+                fatalError("enable permissions in system preferences!")
+            case .authorized:
+                showCamera()
+            default:
+                fatalError("Handle new case for authorization due to deprication")
+            }
+        }
```

Comment:
You may already know this, but It'd probably be a good idea to eventually show an alert (or something) rather than `fatalError` for those cases.